libshpool/src/attach.rs

```diff
@@ -15,13 +15,14 @@
 use std::{env, fmt, io, path::PathBuf, thread, time};
 
 use anyhow::{anyhow, bail, Context};
+use shpool_protocol::{
+    AttachHeader, AttachReplyHeader, ConnectHeader, DetachReply, DetachRequest, ResizeReply,
+    ResizeRequest, SessionMessageReply, SessionMessageRequest, SessionMessageRequestPayload,
+    TtySize,
+};
 use tracing::{error, info, warn};
 
-use super::{
-    config, duration, protocol,
-    protocol::{AttachHeader, ConnectHeader},
-    test_hooks, tty,
-};
+use super::{config, duration, protocol, test_hooks, tty::TtySizeExt};
```

Comment:
(personal preference, feel free to ignore) `use tty::TtySizeExt as _` allows importing a trait into scope for resolution reasons, without being able to reference it by ident directly. Not strictly necessary in this case, but I prefer using this consistently for "method-only" traits.

Reply:
Oh yeah that's a nice pattern. It makes it more explicit what we are up to which is nice because importing traits for methods is pretty magical and a bit of a readability hit IMO.